src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs

```diff
@@ -58,34 +60,30 @@ public void Initialize(ExtensionConfigContext context)
 
             INameResolver nameResolver = context.Config.GetService<INameResolver>();
 
-            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
-
             // Use this if there is no other connection string set.
             _defaultConnectionString = nameResolver.Resolve(AzureWebJobsCosmosDBConnectionStringName);
 
-            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
-
-            IBindingProvider outputProvider = factory.BindToCollector<CosmosDBAttribute, OpenType>(typeof(CosmosDBCollectorBuilder<>), this);
-
-            IBindingProvider outputProviderJObject = factory.BindToCollector<CosmosDBAttribute, JObject>(typeof(CosmosDBCollectorBuilder<>), this);
-
-            IBindingProvider clientProvider = factory.BindToInput<CosmosDBAttribute, DocumentClient>(new CosmosDBClientBuilder(this));
+            // Apply ValidateConnection to all on this rule. 
+            var rule = context.AddBindingRule<CosmosDBAttribute>();
+            rule.AddValidator(ValidateConnection);
+            rule.BindToCollector<OpenType>(typeof(CosmosDBCollectorBuilder<>), this);
+            rule.BindToCollector<JObject>(typeof(CosmosDBCollectorBuilder<>), this);
 
-            IBindingProvider jArrayProvider = factory.BindToInput<CosmosDBAttribute, JArray>(typeof(CosmosDBJArrayBuilder), this);
+            rule.BindToInput<DocumentClient>(new CosmosDBClientBuilder(this));
+            rule.BindToInput<JArray>(typeof(CosmosDBJArrayBuilder), this);
 
-            IBindingProvider enumerableProvider = factory.BindToInput<CosmosDBAttribute, IEnumerable<OpenType>>(typeof(CosmosDBEnumerableBuilder<>), this);
-            enumerableProvider = factory.AddValidator<CosmosDBAttribute>(ValidateInputBinding, enumerableProvider);
+            rule.BindToInput<IEnumerable<OpenType>>(typeof(CosmosDBEnumerableBuilder<>), this).AddValidator(ValidateInputBinding);
```

Comment:
Same semantics here... the validator runs *After* the rule is matched (this is important so that we don't attempt validation on other rules that would obviously fail). So in the fluent interface, it's Bind().AddValidate()